index.js

```diff
@@ -4,17 +4,26 @@ const fp = require('fastify-plugin')
 const { createHash } = require('crypto')
 const fnv1a = require('./fnv1a')
 
-function buildHashFn (algorithm = 'fnv1a') {
-  if (algorithm === 'fnv1a') {
-    return (payload) => '"' + fnv1a(payload).toString(36) + '"'
-  }
+function buildHashFn (algorithm = 'fnv1a', weak = false) {
+  if (weak) {
+    if (algorithm === 'fnv1a') {
+      return (payload) => 'W/"' + fnv1a(payload).toString(36) + '"'
+    }
 
-  return (payload) => '"' + createHash(algorithm)
-    .update(payload).digest().toString('base64') + '"'
+    return (payload) => 'W/"' + createHash(algorithm)
+      .update(payload).digest().toString('base64') + '"'
+  } else {
+    if (algorithm === 'fnv1a') {
+      return (payload) => '"' + fnv1a(payload).toString(36) + '"'
+    }
+
+    return (payload) => '"' + createHash(algorithm)
+      .update(payload).digest().toString('base64') + '"'
+  }
```

Comment:
Can you please refactor this function to avoid code duplications? You can achieve this by using an external variable for the string prefix.

Reply:
Thank you for pointing that out! I've made the requested changes :)